textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -0,0 +1,363 @@
+            throws StorageException {
+        // table should not exist
+        if (checkTableExistence(tableName)) {
+            throw new StorageException(String.format("Table %s already exists.", tableName));
+        }
+        
+        Schema tableSchema = Utils.getSchemaWithID(schema);
+                
+        // write collection catalog
+        DataStore collectionCatalogStore = new DataStore(CatalogConstants.COLLECTION_CATALOG_DIRECTORY,
+                CatalogConstants.COLLECTION_CATALOG_SCHEMA);      
+        insertTupleToDirectory(collectionCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(),
+                CatalogConstants.getCollectionCatalogTuple(tableName, indexDirectory, luceneAnalyzer));
+       
+        // write schema catalog
+        DataStore schemaCatalogStore = new DataStore(CatalogConstants.SCHEMA_CATALOG_DIRECTORY,
+                CatalogConstants.SCHEMA_CATALOG_SCHEMA);
+        for (ITuple tuple : CatalogConstants.getSchemaCatalogTuples(tableName, tableSchema)) {
+            insertTupleToDirectory(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(), tuple);
+        }
+        
+    }
+
+    // drop a table
+    public void deleteTable(String tableName) throws StorageException {
+        Query catalogTableNameQuery = new TermQuery(new Term(CatalogConstants.COLLECTION_NAME, tableName));
+        
+        DataWriter collectionCatalogWriter = new DataWriter(CatalogConstants.COLLECTION_CATALOG_DATASTORE, 
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        collectionCatalogWriter.deleteTuple(catalogTableNameQuery);
+        
+        DataWriter schemaCatalogWriter = new DataWriter(CatalogConstants.SCHEMA_CATALOG_DATASTORE,
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        schemaCatalogWriter.deleteTuple(catalogTableNameQuery);
+    }
+
+    // insert a tuple to a table, returns the ID field
+    public IDField insertTuple(String tableName, ITuple tuple) throws StorageException {
+        if (! checkTableExistence(tableName)) {
+            throw new StorageException(String.format("Table %s doesn't exist.", tableName));
+        }
+        try {
+            DataStore tableDataStore = getTableDataStore(tableName);
+            String tableAnalyzerString = getTableAnalyzer(tableName);
+            Analyzer luceneAnalyzer = LuceneAnalyzerConstants.getLuceneAnalyzer(tableAnalyzerString);
+
+            return insertTupleToDirectory(tableDataStore, luceneAnalyzer, tuple);
+        } catch (DataFlowException e) {
+            throw new StorageException(e);
+        }
+    }
+    
+    private IDField insertTupleToDirectory(IDataStore dataStore, Analyzer luceneAnalyzer, ITuple tuple) throws StorageException {
+        String tableDirectory = dataStore.getDataDirectory();
+        Schema tableSchema = dataStore.getSchema();
+        IDField idField;
+                
+        if (! tableSchema.containsField(SchemaConstants._ID)) {
```

Comment:
Since the `getSchemaWithID()` already checks the existence of the `_id` field, this `if` is not necessary.